content-src/lib/clamp-title-excerpt.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Adjusts the line-clamp of an excerpt to fill desired number of lines.
+ *
+ * This is a React callback ref that should be set on a title node with an
+ * excerpt node as the next sibling -- both nodes should be styled to normally
+ * clamp lines via CSS. The title node needs two data attributes to configure
+ * how many lines the title can be and how many total lines with the excerpt.
+ */
+export function clampTitleExcerpt(title) {
+  if (title) {
+    // Make sure we have enough data to even clamp lines
+    const excerpt = title.nextSibling;
+    const {titleLines, totalLines} = title.dataset;
+    if (excerpt && titleLines && totalLines) {
+      // Set the appropriate clamp for excerpt if title uses fewer lines
+      const lineHeight = global.getComputedStyle(title).lineHeight.slice(0, -2);
```

Comment:
Seems like `parseInt` would be more bulletproof and self-documenting than slicing characters?

Reply:
How did you want to pass the string in to parseInt? A number with "px" results in `NaN`

Reply:
That's odd. If I use the console and do something like `parseInt(getComputedStyle(document.querySelector('h1')).lineHeight)` it works.

Reply:
Hmm really odd. Pretty sure that's what I tried first but got `NaN` so then ended up with `slice`. I do see `parseInt("20px")` returning `20` so okay! I'll switch it

Reply:
I wonder if you might be hitting a race condition with the DOM. That was another concern I was going to raise separately because sometimes the DOM isn't fully ready when refs are linked, at least in regards to doing things like computed style measurements.